src/app/shared/services/dynamic-data/dynamic-data.service.ts

```diff
@@ -109,16 +109,16 @@ export class DynamicDataService extends AsyncServiceBase {
     flow_type: FlowTypes.FlowType,
     flow_name: string,
     row_id: string,
-    update: Partial<T>
+    update: Partial<T>,
+    options: { upsert: boolean } = { upsert: false }
   ) {
     if (update) {
       const { collectionName } = await this.ensureCollection(flow_type, flow_name);
       const existingDoc = await this.db.getDoc<any>(collectionName, row_id);
-
       if (existingDoc) {
         const data = existingDoc.toMutableJSON();
         update = deepMergeObjects(data, update);
-      } else {
+      } else if (!options.upsert) {
```

Comment:
nit(non-blocking)
I think we can probably allow the default behaviour of `update` should be to upsert, remove the option and error (and if required include code to create doc if db update doesn't work)